### PR TITLE
WIP #10764

### LINKED ISSUE #10764: ListView options=>tag=>false doesn't work correctly
Hi all

I need accordion, and use ListView (no Collapse), but ListView insert '<>' berore summary:

recipe:

view.php :
```
<div class="panel-group" id="accordion">
		<?= ListView::widget([
			'dataProvider' => $dataProvider,
			'itemView'     => '_item',
			'id'           => false,
			'options'      => [
				'tag' => false,
			],
			'itemOptions'  => [
				'tag' => false,
			],
		]); ?>
</div>
```

_item.php
```
<div class="panel panel-default">
	<div class="panel-heading">
		<h4 class="panel-title">
.........................................
```
and see in html:

```
<div class="panel-group" id="accordion">
 		<><div class="summary">............ <b>1-3</b> ..... <b>3</b>.</div>
 <div class="panel panel-default">
 	<div class="panel-heading">
 		<h4 class="panel-title">
```

but I expect to see

```
<div class="panel-group" id="accordion">
 		<div class="summary">............ <b>1-3</b> ..... <b>3</b>.</div>
 <div class="panel panel-default">
 	<div class="panel-heading">
 		<h4 class="panel-title">
```

### PR DESCRIPTION
**Pull request by pana1990**
taks : 
- [x] check code
- [x] check docs
- [x] add more tests
- [x] update change log

fixes #10764

### CODE REVIEW

**Comment by lynicidn** [comment]
> why? travis pass all tests.

try set 
```php
'itemOption' => ['tag' => false]
```

**Comment by pana1990** [comment]
@lynicidn https://github.com/pana1990/yii2/commit/f4990b3879b326b651aedec9076cb0df72af3154#diff-f55c5f70280cbbb7a9f1774fdfbde809R119 travis pass test, can you check  it?

**Comment by lynicidn** [comment]
@pana1990 sorry, u right, it will be work :+1:

**Comment by pana1990** [comment]
Thanks :)

**Comment by lynicidn** [comment]
@pana1990 thanks for you =) and again sorry

**Comment by SilverFire** [comment]
Looks good. 
I think we must do the same for `Html::beginTag` and `Html::endTag` to be consistent.

**Comment by pana1990** [comment]
@SilverFire I agree.

**Comment by pana1990** [comment]
@SilverFire done!

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -119,7 +119,7 @@ public static function decode($content)
 
     /**
      * Generates a complete HTML tag.
-     * @param string $name the tag name
+     * @param string|boolean|null $name the tag name. If $name is null or false, the corresponding content will be rendered without any tag.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: `null` -> <code>`null`</code>
`false` -> <code>`false`</code>

File: framework/helpers/BaseHtml.php [hunk 2]
```diff
@@ -137,13 +137,16 @@ public static function decode($content)
...
+            return $content;
+        }
         $html = "<$name" . static::renderTagAttributes($options) . '>';
         return isset(static::$voidElements[strtolower($name)]) ? $html : "$html$content</$name>";
     }
 
     /**
      * Generates a start tag.
-     * @param string $name the tag name
+     * @param string|boolean|null $name the tag name. If $name is null or false, the corresponding content will be rendered without any tag.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: `null` -> <code>\`null\`</code>
`false` -> <code>\`false\`</code>

File: framework/helpers/BaseHtml.php [hunk 3]
```diff
@@ -154,18 +157,24 @@ public static function tag($name, $content = '', $options = [])
...
+        if ($name === null || $name === false) {
+            return '';
+        }
         return "<$name" . static::renderTagAttributes($options) . '>';
     }
 
     /**
      * Generates an end tag.
-     * @param string $name the tag name
+     * @param string|boolean|null $name the tag name. If $name is null or false, the corresponding content will be rendered without any tag.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: `null` -> <code>\`null\`</code>
`false` -> <code>\`false\`</code>

Comment (pana1990) [inline framework/helpers/BaseHtml.php]: @SilverFire done! 

Thanks for your help

**Comment by SilverFire** [comment]
Ready to merge.
TODO: Add a changelog

**Comment by pana1990** [comment]
i will wait for next milestone to update change log :+1:

**Comment by SilverFire** [comment]
sure, thank you

**Comment by pana1990** [comment]
@SilverFire can you suggest me some phrase to changelog?

**Comment by SilverFire** [comment]
<code>\`yii\helpers\Html::tag()\` and \`::beginTag()\` return content without any HTML when the \`$tag\` attribute is \`false\` or \`null\`</code>

**Comment by pana1990** [comment]
@SilverFire done! thanks you very much ;)

**Comment by SilverFire** [comment]
Could you update from the upstream, please?

**Comment by pana1990** [comment]
@SilverFire  done!

**Comment by SilverFire** [comment]
Thank you!

**Pull request merged**